libraries/AP_NavEKF2/AP_NavEKF2_Control.cpp

```diff
@@ -120,7 +121,7 @@ void NavEKF2_core::setWindMagStateLearningMode()
             P[21][21] = bodyMagFieldVar.z;
         } else {
             // set the variances equal to the observation variances
-            for (uint8_t index=18; index<=21; index++) {
+            for (uint8_t index=16; index<=21; index++) {
                 P[index][index] = sq(frontend->_magNoise);
```

Comment:
What are the significances of indices 16 and 21?  Can these magic numbers be #define, const, or const expr??

Reply:
Or at least put a comment explaining why we're going from 16 to 21?